flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneJobClusterEntryPoint.java

```diff
@@ -97,14 +100,27 @@ public static void main(String[] args) {
 
 		StandaloneJobClusterEntryPoint entrypoint = new StandaloneJobClusterEntryPoint(
 			configuration,
-			clusterConfiguration.getJobId(),
+			createJobIdForCluster(clusterConfiguration.getJobId(), configuration),
 			clusterConfiguration.getSavepointRestoreSettings(),
 			clusterConfiguration.getArgs(),
 			clusterConfiguration.getJobClassName());
```

Comment:
If `getJobClassName` returns an `Optional`, then we should insert `clusterConfiguration.getJobClassName().orElse(null)`

Reply:
Its not returning an Optional anymore.